### DISCUSSION TITLE
Search history by more than just command

### DISCUSSION THREAD

**Discussion by alokedesai** [warpdotdev/Warp#635] [Product roadmap]
**Is your feature request related to a problem? Please describe.**
Most shells currently only support searching through your history by the command. This is frustrating because I can't search by the output, the directory, timestamp, the box, or any other metadata. Additionally, the default history file is too small, I often try to search for commands that I ran a few months ago only to find they were flushed from my history.

**Describe the solution you'd like**
I would like to be able to be able to have a UI to infinitely search my terminal history. To start, I'd like to be able to search by:
- Command
- Output
- Timestamp
- Exit code
- Directory
- Box (e.g a specific remote box or a set of remote boxes)

**Additional context**
Once I find a command that matches my search result, I'd like to be able to find commands that I ran before and after in that specific session where the command was executed. Technically, we could do this by indexing Blocks in a local sqllite database or send them to our server and index them in the cloud.

**Comment by elviskahoro** [comment]
There may be some concepts that can be incorporated from Ranger.
A discord comment (feature-requests: 07/14/2021) mentioned elvish shell and their ranger based built in file manager.
![unknown](https://user-images.githubusercontent.com/29553206/132909948-3ab171c5-3a86-433f-8781-3534a7c2d6c8.png)

**Comment by elviskahoro** [comment]
"I mean that just as one can search command history, storing and searching output history would be something that maybe is lacking in other apps. Wouldn’t such be nice to be able to view weeks or more of output and search?"

**Comment by elviskahoro** [comment]
Fromm mantis via Discord: https://discord.com/channels/851854972600451112/914052748608143360/915443415234277406
> Thanks. Perhaps there could be some configurable search (and possibly completion), such that using some particular backend, the presented UI is nicer and as native as possible, e.g. on macOS appearing like the command palette.
> There was some other terminal, I forgot which, that seemed to have useful features. I vaguely recall some SSH manager. Maybe thru use and indexing history, servers could automatically appear in the command palette.
> Concerning the command palette, perhaps you are familar with JetBrains IDEs. Maybe some default search could search anything available, and the UI may have tabs if one wishes to filter some category such as Warp commands, history, servers, completion, etc.

**Comment by neongreen** [comment]
"Show history for this directory" would be a killer. I just navigated to a project and thought "damn, what's the command for building it" (the previous time I built it was several months ago). I'd gladly pay for a monthly subscription to Warp to have this available.
